tests/unit/expire.tcl

```diff
@@ -261,100 +275,270 @@ start_server {tags {"expire"}} {
             {set foo2 bar}
-            {expire foo2 100}
-            {pexpire foo2 100000}
+            {pexpireat foo2 *}
+            {pexpireat foo2 *}
             {set foo3 bar}
-            {expireat foo3 *}
+            {pexpireat foo3 *}
             {pexpireat foo3 *}
             {del foo3}
             {set foo4 bar}
-            {pexpire foo4 200000}
-            {pexpire foo4 200000}
             {pexpireat foo4 *}
             {pexpireat foo4 *}
+            {pexpireat foo4 *}
+            {pexpireat foo4 *}
+            {set foo5 bar}
+            {restore foo6 * {*} ABSTTL}
+            {restore foo7 * {*} absttl}
+        }
+    }
+
+    # Start another server to test replication of TTLs
+    start_server {} {
+        # Set the outer layer server as primary
+        set primary [srv -1 client]
+        set primary_host [srv -1 host]
+        set primary_port [srv -1 port]
+        # Set this inner layer server as replica
+        set replica [srv 0 client]
+
+        test {First server should have role slave after REPLICAOF} {
+            $replica replicaof $primary_host $primary_port
+            wait_for_condition 50 100 {
+                [s 0 role] eq {slave}
+            } else {
+                fail "Replication not started."
+            }
+        }
+
+        test {For all replicated TTL-related commands, absolute expire times are identical on primary and replica} {
+            # Apply each TTL-related command to a unique key on primary
+            # SET commands
+            $primary set foo1 bar ex 100
+            $primary set foo2 bar px 100000
+            $primary set foo3 bar exat [expr [clock seconds]+100]
+            $primary set foo4 bar pxat [expr [clock milliseconds]+100000]
+            $primary setex foo5 100 bar
+            $primary psetex foo6 100000 bar
+            # EXPIRE-family commands
+            $primary set foo7 bar
+            $primary expire foo7 100
+            $primary set foo8 bar
+            $primary pexpire foo8 100000
+            $primary set foo9 bar
+            $primary expireat foo9 [expr [clock seconds]+100]
+            $primary set foo10 bar
+            $primary pexpireat foo10 [expr [clock milliseconds]+100000]
+            # GETEX commands
+            $primary set foo11 bar
+            $primary getex foo11 ex 100
+            $primary set foo12 bar
+            $primary getex foo12 px 100000
+            $primary set foo13 bar
+            $primary getex foo13 exat [expr [clock seconds]+100]
+            $primary set foo14 bar
+            $primary getex foo14 pxat [expr [clock milliseconds]+100000]
+            # RESTORE commands
+            $primary set foo15 bar
+            set encoded [$primary dump foo15]
+            $primary restore foo16 100000 $encoded
+            $primary restore foo17 [expr [clock milliseconds]+100000] $encoded absttl
+
+            # Wait for replica to get the keys and TTLs
+            assert {[$primary wait 1 1000] == 1}
```

Comment:
Waiting for `1000` makes this test sometimes flaky on my slower local machine. Change it to without a timeout to de-flake it.